src/category_theory/adjunction/basic.lean

```diff
@@ -204,6 +204,48 @@ def id : 𝟭 C ⊣ 𝟭 C :=
   unit := 𝟙 _,
   counit := 𝟙 _ }
 
+/-- If F and G are naturally isomorphic functors, establish an equivalence of hom-sets. -/
+@[reducible]
+def equiv_homset_left_of_nat_iso
+  {F G : C ⥤ D} (iso : F ≅ G) {X : C} {Y : D} :
+  (F.obj X ⟶ Y) ≃ (G.obj X ⟶ Y) :=
+{ to_fun := λ f, (iso.app _).inv ≫ f,
+  inv_fun := λ g, (iso.app _).hom ≫ g,
+  left_inv := λ f, by simp,
+  right_inv := λ g, by simp }
+
+/-- If G and H are naturally isomorphic functors, establish an equivalence of hom-sets. -/
+@[reducible]
+def equiv_homset_right_of_nat_iso
+  {G H : D ⥤ C} (iso : G ≅ H) {X : C} {Y : D} :
+  (X ⟶ G.obj Y) ≃ (X ⟶ H.obj Y) :=
+{ to_fun := λ f, f ≫ (iso.app _).hom,
+  inv_fun := λ g, g ≫ (iso.app _).inv,
+  left_inv := λ f, by simp,
+  right_inv := λ g, by simp }
+
+/-- Transport an adjunction along an natural isomorphism on the left. -/
+def of_nat_iso_left
+  {F G : C ⥤ D} {H : D ⥤ C} (adj : F ⊣ H) (iso : F ≅ G) :
+  G ⊣ H :=
+adjunction.mk_of_hom_equiv
+{ hom_equiv := λ X Y, (equiv_homset_left_of_nat_iso iso.symm).trans (adj.hom_equiv X Y) }
+
+/-- Transport an adjunction along an natural isomorphism on the right. -/
+def of_nat_iso_right
+  {F : C ⥤ D} {G H : D ⥤ C} (adj : F ⊣ G) (iso : G ≅ H) :
+  F ⊣ H :=
+adjunction.mk_of_hom_equiv
+{ hom_equiv := λ X Y, (adj.hom_equiv X Y).trans (equiv_homset_right_of_nat_iso iso) }
+
+def right_adjoint_of_nat_iso {F G : C ⥤ D} (h : F ≅ G) [r : is_right_adjoint F] : is_right_adjoint G :=
+{ left := r.left,
+  adj := of_nat_iso_right r.adj h }
+
+def left_adjoint_of_nat_iso {F G : C ⥤ D} (h : F ≅ G) [r : is_left_adjoint F] : is_left_adjoint G :=
```

Comment:
Could be an instance?

Reply:
Or maybe it'd fail since `h` wouldn't be inferred?

Reply:
Yes, it won't get anywhere without the `h`. Just leave this as a `def`.